### DISCUSSION TITLE
ike response received but StrongSwan not processing

### DISCUSSION THREAD

**Discussion by stourmarine** [strongswan/strongswan#1268] [Q&A]
Hi,
I have same StrongSwan version installed in two servers, one is working ok, one is getting timeout
# strongswan --version
Linux strongSwan U5.7.2/K5.10.135-122.509.amzn2.x86_64
University of Applied Sciences Rapperswil, Switzerland

Sep  1 13:40:37 strongswan2 charon: 08[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:40:41 strongswan2 charon: 07[IKE] retransmit 1 of request with message ID 0
Sep  1 13:40:41 strongswan2 charon: 07[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:40:48 strongswan2 charon: 09[IKE] retransmit 2 of request with message ID 0
Sep  1 13:40:48 strongswan2 charon: 09[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:41:01 strongswan2 charon: 11[IKE] retransmit 3 of request with message ID 0
Sep  1 13:41:01 strongswan2 charon: 11[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:41:24 strongswan2 charon: 06[IKE] retransmit 4 of request with message ID 0
Sep  1 13:41:24 strongswan2 charon: 06[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:42:06 strongswan2 charon: 12[IKE] retransmit 5 of request with message ID 0
Sep  1 13:42:06 strongswan2 charon: 12[NET] sending packet: from 10.0.x.x[500] to y.y.y.y[500] (1300 bytes)
Sep  1 13:43:22 strongswan2 charon: 08[IKE] giving up after 5 retransmits
Sep  1 13:43:22 strongswan2 charon: 08[IKE] establishing IKE_SA failed, peer not responding

I can see the peer responding to each request:
11:40:49.243490 Out 0a:1f:e4:5c:c2:22 ethertype IPv4 (0x0800), length 1344: 10.0.x.x.isakmp > y.y.y.y.isakmp: isakmp: parent_sa ikev2_init[I]
11:40:49.255810  In 0a:76:d3:cb:e2:b8 ethertype IPv4 (0x0800), length 516: y.y.y.y.isakmp > 10.0.x.x.isakmp: isakmp: parent_sa ikev2_init[R]
11:40:49.262727  In 0a:76:d3:cb:e2:b8 ethertype IPv4 (0x0800), length 516: y.y.y.y.isakmp > 10.0.x.x.isakmp: isakmp: parent_sa ikev2_init[R]
11:41:31.229525 Out 0a:1f:e4:5c:c2:22 ethertype IPv4 (0x0800), length 1344: 10.0.x.x.isakmp > y.y.y.y.isakmp: isakmp: parent_sa ikev2_init[I]
11:41:31.233751 Out 0a:1f:e4:5c:c2:22 ethertype IPv4 (0x0800), length 1344: 10.0.x.x.isakmp > y.y.y.y.isakmp: isakmp: parent_sa ikev2_init[I]
11:41:31.247469  In 0a:76:d3:cb:e2:b8 ethertype IPv4 (0x0800), length 516: y.y.y.y.isakmp > 10.0.x.x.isakmp: isakmp: parent_sa ikev2_init[R]

What could it be wrong?

Best Regards,
Mihai Radulescu

**Comment by Thermi** [comment]
iptables or nftables could be dropping the responses or the packets (requests or responses) are too large and can't be transmitted. Try enabling IKE fragmentation on both sides.

**Reply by stourmarine** [reply]
Thanks for answering this, but it's not the case:
#> iptables-save
# Generated by iptables-save v1.8.4 on Mon Sep  5 08:35:24 2022
*filter
:INPUT ACCEPT [39959:81099974]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [36024:2821048]
COMMIT
# Completed on Mon Sep  5 08:35:24 2022
#> nft list table filter
-bash: nft: command not found

**Reply by Thermi** [reply]
yeah then probably a fragmentation issue where the response is too large and fragments get lost. Try enabling IKE fragmentation (fragmentation=yes on both sides)

**Comment by tobiasbrunner** [comment]
This could happen if there are multiple IKE daemons running and listening on UDP port 500.

**Comment by Thermi** [comment]
Hmmmhhhh is there a good reason for not aborting initialization of the daemon when the port can't be exclusive opened?

Am 6. September 2022 07:02:40 UTC schrieb Tobias Brunner ***@***.***>:
>This could happen if there are multiple IKE daemons running and listening on UDP port 500.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/strongswan/strongswan/discussions/1268#discussioncomment-3566452
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from mobile. Please excuse any typos or brevity

**Reply by tobiasbrunner** [reply]
Hm, good question. `SO_REUSEADDR` has been set on the sockets from the beginning. In early versions of strongSwan that actually was necessary due to the dual-daemon architecture. While charon used a raw socket to receive messages that, otherwise, only pluto would have received via its previously bound sockets, it still used regular sockets bound to the same ports to send messages. I guess, since the removal of pluto there is technically no reason to set the option on these UDP sockets anymore (for TCP there would be the potential `TIME_WAIT` issue when restarting, but that's not a concern for UDP). However, the way it currently is implemented, a failure to open/bind any sockets doesn't cause a shutdown of the daemon. But sending would fail with an error that no socket implementation is available, so at least it would not result in this behavior. You can try the _socket-no-reuseaddr_ branch if you like.

**Reply by Thermi** [reply]
Works, but we still should fail initialization. Otherwise people will overlook that.

**Comment by stourmarine** [comment]
Thank you all for the answer. I have wiped out and reinstalled the server from scratch. The tunnels connected instantly. I assume we would never know what was the issue.
